IndomitableIbises/Bangazon-API-Sprint-1
# Payment type model and viewer

## Link to Ticket
Closes #3 

## Description of Proposed Changes

You can now add different payment types with account numbers to a customer.

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout CR-Payment-Type
```

1.  First step is to run django_data.sh API faker_factory
1. Double check that the tables have formed correctly in sqlite
1. Then load up python server
1. You should see the option for the extension to payment type
1. You have to make a few customers first as you must have a customer to attach a payment type (dur 😆 ) {Faker should now do this for you automatically}
1. Once there are customers go in to the payment type local host extension and fill out all the data, attaching them to customers via their customer id number
1. You should still be able to make adjustments and delete payment types once they have been set up
1. Double check in the ERD that all tables are there and populated accordingly
1. Code should be commented with author and explanations

(Please note that since this uses a foreign key it can not be auto-populated with faker data, BUT the customers can!)


## Impacted Areas in Application

List general components of the application that this PR will affect:

* Customer's extension table
* Payment type table made

## Mentions @username

@jcarter0149 @rafcevallos @sirwin4 

## Definition of Done

1. The project must be fully documented. This includes the following:
   1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
   1. Every class must be documented with purpose, author, and methods.
   1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each
